### PR TITLE
HIVE-24861: Hive JDBC driver doesn't consider the value of 'hasMoreRows'

### PR DESCRIPTION
**Pull request by dengzhhu653**

### CODE REVIEW

File: itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestHiveQueryResultSet.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      hiveConf.setBoolean("hive.support.concurrency", false);
+      this.emitEmptyRows = hiveConf.getBoolean(EMIT_EMPTY_ROWS, false);
+      this.numRows = hiveConf.getInt(EMIT_NUM_ROWS, 10);
+      this.emptyRowPos = (numRows / 2);
+      this.position = 0;
+      super.init(hiveConf);
+    }
+
+    @Override
+    public TFetchResultsResp FetchResults(TFetchResultsReq req) throws TException {
```
Comment (ellieMayVelasquez) [inline itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestHiveQueryResultSet.java:69]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Comment (dengzhhu653) [reply itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestHiveQueryResultSet.java:69]: Thanks for the comments, @ellieMayVelasquez! The method is defined the thrift, and used for many other languages: php, python, etc. It's may difficult to change the method.

Comment (ellieMayVelasquez) [reply itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestHiveQueryResultSet.java:69]: noticed.

**Comment by zmh1230** [comment]
Why is this closed? I recently encountered two problems using jdbc to connect to impala. One scenario is that the interval between two batches of data exceeds FETCH_ROWS_TIMEOUT_MS, and one scenario is that the row size exceeds MAX_ROW_SIZE.

**Comment by dengzhhu653** [comment]
> Why is this closed? I recently encountered two problems using jdbc to connect to impala. One scenario is that the interval between two batches of data exceeds FETCH_ROWS_TIMEOUT_MS, and one scenario is that the row size exceeds MAX_ROW_SIZE.

This PR is automatically closed by the PR stale checker for no recent activity within some time. From my understanding, exceeding the FETCH_ROWS_TIMEOUT_MS(I'm not sure Hive JDBC uses this) or MAX_ROW_SIZE should not affect the rows returned, the issue here is that the HiveServer2 always sets the `hasMoreRows` to false:
https://github.com/apache/hive/blob/master/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java#L905,
it could be a problem when the client determines whether the stream reaches its end or not by this variable.

**Pull request closed without merging**